python/distance_calculator.py

```diff
@@ -0,0 +1,112 @@
+import mgp
+import math
+import itertools
+
+KM_MULTIPLIER = 0.001
+LATITUDE = 'lat'
+LONGITUDE = 'lng'
+valid_metrics = ['m', 'km']
+
+@mgp.read_proc
+def single(context: mgp.ProcCtx,
+              start: mgp.Nullable[mgp.Vertex],
+              end: mgp.Nullable[mgp.Vertex],
+              metrics: str = 'm'
+              ) -> mgp.Record(distance=mgp.Number):
+    '''
+    This distance calculator procedure  for one entry returns 1 field.
+
+      * `distance` is the final result obtained by calculating distance (in metres) between the 2 points who
+        each have its latitude and longitude.
+
+    The procedure can be invoked in openCypher using the following calls:
+      CALL distance_calculator.single((:Point {lat:1, lng:2}), (:Point {lat:2, lng:3.5})) YIELD distance;
+      MATCH (n1:Point), (n2:Point) CALL distance_calculator.procedure(n, 1) YIELD * RETURN *;
+    '''
+
+    distance = calculate_distance_between_points(
+        dict(start.properties.items()),
+        dict(end.properties.items()),
+        metrics)
+
+    return mgp.Record(distance=distance)
+
+
+@mgp.read_proc
+def multiple(context: mgp.ProcCtx,
+              start_points: list,
+              end_points: list,
```

Comment:
For all complex types, it is better to use typing objects from `typing` module. E.g (maybe this could be useful for Field Notion):

```python
from typing import List, Dict, Union, Optional, Set, Tuple, Iterable
from myobject import MyObject

list_of_str: List[str] = ['a', 'b', 'c']
set_of_ints: Set[int] = {1, 2, 3}
tuple_of_str_list_str: Tuple[str, List[str]] = ('Mrma', ['a', 'b'])
iterable_of_strs: Iterable[str] = ('a', 'b', 'c')

dict_of_str_to_set_int: Dict[str, Set[str]] = { 'a': {1, 2, 3}, 'b': {4, 5, 6} }
dict_of_str_to_my_object: Dict[str, MyObject] = { 'a': MyObject(), 'b': MyObject() }

list_of_strs_and_none: List[Optional[str]] = ['a', None, 'b', 'c'] # Same as List[Union[str, None]]
```

Reply:
So in this case, you could have `List[mgp.Nullable[mgp.Vertex]]` which is more correct typing.

Reply:
it seems that List[mgp.Nullable[mgp.Vertex]] does not longer work so I specified List[mgp.Vertex] and it is working like before